### DISCUSSION TITLE
Since 9.0.0 my sticks don't work.

### DISCUSSION THREAD

**Discussion by StreetCube** [zwave-js/zwave-js#5327] [Q&A] [answered]
I'm using the JS Api directly and recently I tried to upgrade from 8.x.x versions to 10.x.x versions. Something hadn't work with the driver in the application so I downgraded to 9.x.x but no improvement at all. I thought something changed with the implementation maybe so for test purposes I created a separate application that only does this: 
![image](https://user-images.githubusercontent.com/79054804/212297541-3adc3802-af4c-4006-99bd-360ae98f2c48.png)
(this is exactly your quick start guide basically)

Somehow my driver can't send response back, or if it does it's "unexpected":

11:19:50.841 DRIVER   ███████╗ ██╗    ██╗  █████╗  ██╗   ██╗ ███████╗             ██╗ ███████╗
                      ╚══███╔╝ ██║    ██║ ██╔══██╗ ██║   ██║ ██╔════╝             ██║ ██╔════╝
                        ███╔╝  ██║ █╗ ██║ ███████║ ██║   ██║ █████╗   █████╗      ██║ ███████╗
                       ███╔╝   ██║███╗██║ ██╔══██║ ╚██╗ ██╔╝ ██╔══╝   ╚════╝ ██   ██║ ╚════██║
                      ███████╗ ╚███╔███╔╝ ██║  ██║  ╚████╔╝  ███████╗        ╚█████╔╝ ███████║
                      ╚══════╝  ╚══╝╚══╝  ╚═╝  ╚═╝   ╚═══╝   ╚══════╝         ╚════╝  ╚══════╝
11:19:50.847 DRIVER   version 10.3.1
11:19:50.848 DRIVER   
11:19:50.848 DRIVER   starting driver...
11:19:50.864 DRIVER   opening serial port COM3
11:19:50.877 DRIVER   serial port opened
11:19:50.879 SERIAL » [NAK]                                                                   (0x15)
11:19:50.884 DRIVER   loading configuration...
11:19:50.889 CONFIG   version 10.3.1
11:19:54.919 CONFIG   Device index regenerated
11:19:54.941 DRIVER   beginning interview...
11:19:54.942 DRIVER   added request handler for AddNodeToNetwork (0x4a)...
                      1 registered
11:19:54.944 DRIVER   added request handler for RemoveNodeFromNetwork (0x4b)...
                      1 registered
11:19:54.946 DRIVER   added request handler for ReplaceFailedNode (0x63)...
                      1 registered
11:19:54.947 CNTRLR   querying controller IDs...
11:19:55.040 SERIAL » 0x01030020dc                                                         (5 bytes)
11:19:55.042 DRIVER » [REQ] [GetControllerId]
11:19:55.044 SERIAL « [ACK]                                                                   (0x06)
11:19:55.050 SERIAL « 0x01080120dc0cc5d50117                                              (10 bytes)
11:19:55.052 SERIAL » [ACK]                                                                   (0x06)
11:19:55.054 DRIVER « [RES] [GetControllerId]
                        home ID:     0xdc0cc5d5
                        own node ID: 1
11:19:55.057 DRIVER « [RES] [GetControllerId]                                           [unexpected]
                        home ID:     0xdc0cc5d5
                        own node ID: 1
11:19:55.058 DRIVER   unexpected response, discarding...
11:19:56.054 CNTRLR   Failed to execute controller command after 1/3 attempts. Scheduling next try i
                      n 100 ms.
11:19:56.157 SERIAL » 0x01030020dc                                                         (5 bytes)
11:19:56.157 DRIVER » [REQ] [GetControllerId]
11:19:56.159 SERIAL « [ACK]                                                                   (0x06)
11:19:56.161 SERIAL « 0x01080120dc0cc5d50117                                              (10 bytes)
11:19:56.162 SERIAL » [ACK]                                                                   (0x06)
11:19:56.164 DRIVER « [RES] [GetControllerId]
                        home ID:     0xdc0cc5d5
                        own node ID: 1
11:19:56.166 DRIVER « [RES] [GetControllerId]                                           [unexpected]
                        home ID:     0xdc0cc5d5
                        own node ID: 1
11:19:56.167 DRIVER   unexpected response, discarding...
11:19:57.161 CNTRLR   Failed to execute controller command after 2/3 attempts. Scheduling next try i
                      n 1100 ms.
11:19:58.263 SERIAL » 0x01030020dc                                                         (5 bytes)
11:19:58.264 DRIVER » [REQ] [GetControllerId]
11:19:58.265 SERIAL « [ACK]                                                                   (0x06)
11:19:58.267 SERIAL « 0x01080120dc0cc5d50117                                              (10 bytes)
11:19:58.268 SERIAL » [ACK]                                                                   (0x06)
11:19:58.269 DRIVER « [RES] [GetControllerId]
                        home ID:     0xdc0cc5d5
                        own node ID: 1
11:19:58.272 DRIVER « [RES] [GetControllerId]                                           [unexpected]
                        home ID:     0xdc0cc5d5
                        own node ID: 1
11:19:58.273 DRIVER   unexpected response, discarding...
11:19:59.272 DRIVER   Failed to initialize the driver: ZWaveError: Timeout while waiting for an ACK 
                      from the controller (ZW0200)
                          at Driver.sendMessage (C:\Users\userName\Desktop\Sticktest\node_modules
                      \zwave-js\src\lib\driver\Driver.ts:3990:23)
                          at ZWaveController.identify (C:\Users\userName\Desktop\Sticktest\node_m
                      odules\zwave-js\src\lib\controller\Controller.ts:713:33)
                          at Driver.initializeControllerAndNodes (C:\Users\userName\Desktop\Stick
                      test\node_modules\zwave-js\src\lib\driver\Driver.ts:1208:26)
                          at Immediate.<anonymous> (C:\Users\userName\Desktop\Sticktest\node_modu
                      les\zwave-js\src\lib\driver\Driver.ts:1015:16)
ZWaveError: Failed to initialize the driver: ZWaveError: Timeout while waiting for an ACK from the controller (ZW0200)

Everything works fine with versions 8.x.x. I tried to set the timeout property back to the original that didn't work. I can only think the serialport upgrade caused something. I could've missed something with the implementation but at this point I doubt that's the case.

I also tried with different environment, different laptop, different stick. Every time works with 8.x.x but stops working at 9.0.0. Can you please review my problem?

**Comment by AlCalzone** [comment, marked as answer]
See #5024. I've narrowed it down to a dependency update, but not sure why that's the case. The problem happens in an unrelated code part.
